### PR TITLE
feat: support inverting header matches for rate limit

### PR DESCRIPTION
**Pull request by rudrakhp**
**What type of PR is this?**

Feature changes for supporting inverting header matches for rate limiting. 

**What this PR does / why we need it**:

Implements API changes proposed in #4176 

**Which issue(s) this PR fixes**:

Ref #2193

### CODE REVIEW

**Comment by arkodg** [comment]
the PR looks good, can you add some tests in https://github.com/envoyproxy/gateway/tree/main/internal/gatewayapi/testdata & https://github.com/envoyproxy/gateway/tree/main/internal/xds/translator/testdata

File: internal/gatewayapi/backendtrafficpolicy.go [hunk 1]
```diff
@@ -721,12 +722,14 @@ func buildRateLimitRule(rule egv1a1.RateLimitRule) (*ir.RateLimitRule, error) {
...
 					Name:      header.Name,
 					SafeRegex: header.Value,
+					Invert:    header.Invert,
 				}
 				irRule.HeaderMatches = append(irRule.HeaderMatches, m)
 			case *header.Type == egv1a1.HeaderMatchDistinct && header.Value == nil:
 				m := &ir.StringMatch{
 					Name:     header.Name,
 					Distinct: true,
+					Invert:   header.Invert,
```
Comment (arkodg) [inline internal/gatewayapi/backendtrafficpolicy.go:732]: invert doesnt make sense for `Distinct` . We should probably catch that and flag that in  [either in API CEL Validation](https://github.com/envoyproxy/gateway/blob/ffe029b576eef5a500058bbad3b461ac489083a3/api/v1alpha1/ratelimit_types.go#L150) or [Status](https://github.com/envoyproxy/gateway/blob/ffe029b576eef5a500058bbad3b461ac489083a3/internal/gatewayapi/backendtrafficpolicy.go#L692)

Comment (rudrakhp) [reply internal/gatewayapi/backendtrafficpolicy.go:732]: @arkodg Added a check in Status and a corresponding test as well.

Comment (arkodg) [reply internal/gatewayapi/backendtrafficpolicy.go:736]: thanks, can you also rm L736 since its NA ?

Comment (rudrakhp) [reply internal/gatewayapi/backendtrafficpolicy.go]: Done!

**Review by arkodg — APPROVED** [review]
LGTM thanks !

would be great to see some e2e and docs in follow up PRs

**Comment by zirain** [comment]
can you make linter happy?

**Review by zhaohuabing — DISMISSED** [review]

**Comment by arkodg** [comment]
> can you make linter happy?

here are the errors
```

internal/xds/translator/testdata/in/xds-ir/ratelimit.yaml
  Error: 74:13 [indentation] wrong indentation: expected 10 but found 12
  Error: 75:17 [indentation] wrong indentation: expected 14 but found 16
  Error: 86:9 [indentation] wrong indentation: expected 6 but found 8
  Error: 87:13 [indentation] wrong indentation: expected 10 but found 12
  Error: 88:26 [new-line-at-end-of-file] no new line character at the end of file
  ```

**Comment by rudrakhp** [comment]
@arkodg @zirain done!

**Review by zhaohuabing — APPROVED** [review]

**Review by arkodg — APPROVED** [review]

**Pull request merged**